pkg/app/api/grpcapi/api.go

```diff
@@ -176,6 +178,41 @@ func (a *API) ListApplications(ctx context.Context, req *apiservice.ListApplicat
 	}
 
 	const limit = 10
+
+	envFilters := []datastore.ListFilter{
+		{
+			Field:    "ProjectId",
+			Operator: "==",
+			Value:    key.ProjectId,
+		},
+	}
+	if req.EnvId != "" {
+		envFilters = append(envFilters, datastore.ListFilter{
+			Field:    "Id",
+			Operator: "==",
+			Value:    req.EnvId,
+		})
+	}
+	if req.EnvName != "" {
+		envFilters = append(envFilters, datastore.ListFilter{
+			Field:    "Name",
+			Operator: "==",
+			Value:    req.EnvName,
+		})
+	}
+	var envs []*model.Environment
+	// only query for environments list in case EnvName or EnvId is set on request.
+	if req.EnvName != "" || req.EnvId != "" {
+		envListOpts := datastore.ListOptions{
+			Filters:  envFilters,
+			PageSize: limit,
+		}
+		envs, err = listEnvironments(ctx, a.environmentStore, envListOpts, a.logger)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Some points around here I think should be improved:
- `listEnvironments` should be called only if `req.EnvId == ""` && `req.EnvName != ""`
- use `in` operator only if the final list of EnvID (returned from `listEnvironments` or `req.EnvId`) has at least 2 elements

Reply:
To clarify, I want to confirm the behavior of this ListApplications function in case __both env-id and env-name values are passed via pipectl command__. Will we return 
1. only application that matches `env-id` and `env-name` (just one application).
2. all applications that match `env-id` or `env-name`.

In case of (1), the second improvement could be simplified to: in case of not pass `env-id` via command, check for envsID returned from `listEnvironments`. How do you think about it?👀

Reply:
Yes. (1) looks good. All of the flags are ANDed.

Reply:
Thank you, will implement follow (1) 🙏